fe/fe-core/src/main/java/org/apache/doris/catalog/Type.java

```diff
@@ -46,12 +46,8 @@
  * as abstract methods that subclasses must implement.
  */
 public abstract class Type {
-    // Maximum nesting depth of a type. This limit was determined experimentally by
-    // org.apache.doris.rewrite.FoldConstantsRule.apply generating and scanning
-    // deeply nested Parquet and Avro files. In those experiments, we exceeded
-    // the stack space in the scanner (which uses recursion for dealing with
-    // nested types) at a nesting depth between 200 and 300 (200 worked, 300 crashed).
-    public static int MAX_NESTING_DEPTH = 2;
+    // Currently only support Array type with max 9 depths.
+    public static int MAX_NESTING_DEPTH = 9;
```

Comment:
Why is 9?

Reply:
there is a hard limit on BE side. See `TypeInfo* get_array_type_info` in `be/src/olap/types.cpp`:
```
DCHECK(iterations <= 8) << "the depth of nested array type should not be larger than 8";
```